### PR TITLE
fix(pep440): Handle widening ranges with '~=' operator

### PR DESCRIPTION
**Pull request by scemily13**
## Changes

Update pep440 versioning to handle dependency ranges with the `~=` operator when the versioning strategy is `widen`. 

## Context

Renovate is not correctly widening dependency ranges with the `~=` operator. 

Current functionality (e.g. new dependency version of `2.0.0`): 
Old range: `~=1.1.0` 
New range: `~=2.0.0`

Correct functionality:
Old range: `~=1.1.0`
New range: `>=1.1.0,<3`

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/modules/versioning/pep440/range.ts [hunk 1]
```diff
@@ -85,7 +85,7 @@ function getFutureVersion(
 
 interface Range {
   operator: string;
-  prefix: string;
+  prefix?: string;
```
Comment (viceice) [inline lib/modules/versioning/pep440/range.ts:88]: why this change?

Comment (scemily13) [reply lib/modules/versioning/pep440/range.ts:88]: This overall change includes converting ranges with the operator `~=` down into a pair of new range objects, and I didn't feel like it was necessary to include prefixes in either of those objects (see lines 377-381).

Comment (scemily13) [reply lib/modules/versioning/pep440/range.ts:88]: This [already present line](https://github.com/renovatebot/renovate/pull/15704/files#diff-f6e993a4715dd7526818c5bb2cdfdaa3a139db2aa290ba8857b9d93f8fe03c4fL290) suggests that the field should be nullable as well.

Comment (viceice) [reply lib/modules/versioning/pep440/range.ts:88]: ok

**Review by viceice — APPROVED** [review]
